lib/samlp.js

```diff
@@ -1,17 +1,30 @@
 var saml20                = require('saml').Saml20;
 var zlib                  = require('zlib');
 var xmldom                = require('xmldom');
+var crypto                = require('crypto');
 var SignedXml             = require('xml-crypto').SignedXml;
 var xpath                 = require('xpath');
 var xtend                 = require('xtend');
+var querystring           = require('querystring');
 
 var templates             = require('./templates');
 var encoders              = require('./encoders');
 var PassportProfileMapper = require('./claims/PassportProfileMapper');
 
-function getSamlRequest(samlRequest, callback) {
-  if (!samlRequest) return callback();
+var signaturePath = "//*[local-name(.)='AuthnRequest']/*[local-name(.)='Signature' and namespace-uri(.)='http://www.w3.org/2000/09/xmldsig#']";
+var algorithmSignatures = {
+  'http://www.w3.org/2001/04/xmldsig-more#rsa-sha256': 'rsa-sha256',
+  'http://www.w3.org/2000/09/xmldsig#rsa-sha1': 'rsa-sha1'
+};
+
+function getSamlRequest(samlRequest, options, callback) {
```

Comment:
Seems weird to me that a function called `getSamlRequest` receives a parameter called `samlRequest`. What is the function doing? Parsing? Validating?